File: BinaryTree.java
class BinaryTree {

    Node root = null;

    class Node {
        int val;
        Node left, right;

        Node(int val) {
            this.val = val;
            this.left = null;
            this.right = null;
        }
    }

    BinaryTree(int val) {

        root = new Node(val);
    }

    public void inserAtLeft(Node r, int val) {
        Node newNode = new Node(val);
        r.left = newNode;
    }

    public void inserAtRight(Node r, int val) {
        Node newNode = new Node(val);
        r.right = newNode;
    }

    public static void preOrder(Node r) {
        if (r != null) {
            System.out.print(r.val + " ");
            preOrder(r.left);
            preOrder(r.right);
        }
    }

    public static void inOrder(Node r) {
        if (r != null) {
            inOrder(r.left);
            System.out.print(r.val + " ");
            inOrder(r.right);
        }
    }

    public static void postOrder(Node r) {
        if (r != null) {
            postOrder(r.left);
            postOrder(r.right);
            System.out.print(r.val + " ");
        }
    }
}

class Main {
    public static void main(String[] args) {
        BinaryTree tree = new BinaryTree(10);
        tree.inserAtRight(tree.root, 15);
        tree.inserAtLeft(tree.root, 5);
        tree.inserAtLeft(tree.root.left, 3);
        tree.inserAtRight(tree.root.left, 8);
        tree.inserAtLeft(tree.root.right, 12);
        BinaryTree.preOrder(tree.root);
        System.out.println();
        BinaryTree.inOrder(tree.root);
        System.out.println();
        BinaryTree.postOrder(tree.root);

    }
}

File: Binarytree.c
// You are using GCC
#include<stdio.h>
#include<stdlib.h>

struct node{
    int val;
    struct node *left,*right;
};

struct node *root=NULL;

struct node *create(int val){
    struct node *ptr=(struct node *)malloc(sizeof(struct node));
    ptr->val=val;
    ptr->left=NULL;
    ptr->right=NULL;
    return ptr;
}
struct node *insert(struct node *r,int val){
    if(r==NULL){
        return create(val);
    }
    else if(r->val<val){
        r->left=insert(r->left,val);
    }
    else{
        r->right=insert(r->right,val);
    }
    return r;
}


struct node *preOrder(struct node*r){
    if(r!=NULL){
    printf("%d ",r->val);
    preOrder(r->left);
    preOrder(r->right);}
}
struct node *inOrder(struct node*r){
    if(r!=NULL){
        
    inOrder(r->left);
    printf("%d ",r->val);
    inOrder(r->right);
    }
}
struct node *postOrder(struct node*r){
    if(r!=NULL){
        
    postOrder(r->left);
    postOrder(r->right);
    printf("%d ",r->val);
    }
}


int main(){
    int n;
    scanf("%d",&n);
    for(int i=0;i<n;i++){
        int val;
        scanf("%d",&val);
        root=insert(root,val);
    }
    preOrder(root);
    printf("\n");
    inOrder(root);
    printf("\n");
    postOrder(root);
    printf("\n");
}


